garrysmod/lua/includes/deprecation.lua

```diff
@@ -0,0 +1,74 @@
+local function DEPRECATE( f, name, alternative )
+
+	local error_locations = {}
+	local format = "[ERROR] %s:%d: calling deprecated function '%s'\n"
+
+	if ( alternative ) then
+		format = "[ERROR] %s:%d: calling deprecated function '%s' (use '%s')\n"
+	end
+
+	return function( ... )
+
+		local info = debug.getinfo( 2, "Sl" )
+
+		if info then
+
+			local source, currentline = info.short_src, info.currentline
+
+			if ( not error_locations[source .. currentline] ) then
+
+				error_locations[source .. currentline] = true
+				ErrorNoHalt( string.format( format, source, currentline, name, alternative ) )
+
+			end
+
+		end
+
+		return f( ... )
+
+	end
+
+end
+
+
+local ENTITY = FindMetaTable( "Entity" )
+local PLAYER = FindMetaTable( "Player" )
+
+IncludeCS 									= DEPRECATE( IncludeCS, "IncludeCS", "include & AddCSLuaFile" )
+
+math.Dist 									= DEPRECATE( math.Dist, "math.Dist", "math.Distance" )
+math.Max 									= DEPRECATE( math.Max, "math.Max", "math.max" )
+math.Min 									= DEPRECATE( math.Min, "math.Min", "math.min" )
+math.mod 									= DEPRECATE( math.mod, "math.mod", "math.fmod" )
+
+util.tobool 								= DEPRECATE( util.tobool, "util.tobool", "tobool" )
+util.TraceEntityHull 						= DEPRECATE( util.TraceEntityHull, "util.TraceEntityHull" )
```

Comment:
Just looked on the wiki i wonder if they could fix util.TraceEntityHull it could become useful!